### ISSUE TITLE
[BUG] E1050 false positive on SSM dynamic references containing hyphens due to unescaped regex range

### ISSUE THREAD

**Issue by mchittineni** [aws-cloudformation/cfn-lint#4692]
### CloudFormation Lint Version
v1.56.2

### What operating system are you using?
macOS / Linux / Windows

### Describe the bug
When using dynamic references to AWS Systems Manager Parameter Store parameters whose path contains a hyphen (`-`) (e.g. `{{resolve:ssm:/my-app/config}}`), `cfn-lint` reports a false-positive `E1050` validation error:
```text
E1050 Dynamic reference 'resolve:ssm:/my-app/config' to SSM parameters are not allowed here
```

#### Root Cause
In `src/cfnlint/rules/functions/DynamicReference.py` line 25:
https://github.com/aws-cloudformation/cfn-lint/blob/a4ed5bb8bba23101ff2a3b04c8ba5a3a290d20ef/src/cfnlint/rules/functions/DynamicReference.py#L21-L30

```python
_ssm = {
    "prefixItems": [
        {"type": "string", "const": "resolve"},
        {"type": "string", "enum": ["ssm", "ssm-secure"]},
        {"type": "string", "pattern": "[a-zA-Z0-9_.-/]+"},  # <--- Unescaped range .-/
        {"type": "string", "pattern": "\\d+"},
    ],
    "maxItems": 4,
    "type": "array",
}
```

In a regular expression character class, an unescaped `-` between two characters creates an ASCII character range. Here, `.-/` defines the character range from `.` (ASCII 46) to `/` (ASCII 47). 
The hyphen character (`-`) has ASCII value 45, which is **not** included in that range:
```python
>>> import re
>>> bool(re.fullmatch(r"[a-zA-Z0-9_.-/]+", "/my-app/config"))
False
>>> bool(re.fullmatch(r"[a-zA-Z0-9_.-/]+", "-"))
False
```

In contrast, `src/cfnlint/helpers.py` line 162 correctly escaped the hyphen:
```python
REGEX_DYN_REF_SSM = re.compile(r"^.*{{resolve:ssm:[a-zA-Z0-9_\.\-/]+(:\d+)?}}.*$")
```

Because hyphens are standard kebab-case separators in SSM parameter names and paths across AWS, all valid templates using hyphenated SSM parameters fail validation.

### Expected behavior
Dynamic references to SSM parameters containing hyphens (e.g. `{{resolve:ssm:/my-app/db-password}}`) should pass `E1050` validation without error.

### Reproduction template
```yaml
AWSTemplateFormatVersion: "2010-09-09"
Resources:
  MyQueue:
    Type: AWS::SQS::Queue
    Properties:
      QueueName: "{{resolve:ssm:/my-app/queue-name}}"
```

Command:
```bash
cfn-lint template.yaml
```

Actual output:
```text
E1050 Dynamic reference 'resolve:ssm:/my-app/queue-name' to SSM parameters are not allowed here
template.yaml:6:7
```

### Proposed Fix
In `src/cfnlint/rules/functions/DynamicReference.py` line 25, escape the hyphen:
```diff
-        {"type": "string", "pattern": "[a-zA-Z0-9_.-/]+"},
+        {"type": "string", "pattern": "[a-zA-Z0-9_.\\-/]+"},
```
